File: arrays_and_strings/check_if_palindrome.java
public boolean checkIfPalindrome(String s)
{
    int left = 0;
    int right = s.length() - 1;

    while (left < right)
    {
        if (s.charAt(left) != s.charAt(right))
        {
            return false;
        }

        left++;
        right--;
    }

    return true;
}

File: arrays_and_strings/check_if_palindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */

/**
 * Alternatively can use ES6 arrow function,
 * let checkIfPalindrome = (s) => { ... }
 */
var checkIfPalindrome = function(s) {
    let left = 0;
    let right = s.length - 1;

    while (left < right) {
        if (s[left] != s[right]) {
            return false;
        }

        left++;
        right--;
    }

    return true;
}